java/sketch-editor/README.md

```diff
@@ -1,30 +1,37 @@
-# Sketch Editor
-The Sketch Editor can be used to draw on a map.
-
-![Sketch Editor App](sketch-editor.png)
-
-# How to use the sample
-Use the buttons along the bottom of the screen to sketch: points, multipoints, polylines, polygons, freehand lines, and freehand polylines.
-
-Use the buttons at the top of the screen to: undo, redo, and stop (finish) the sketch.
-
-# How it works
-1. Create a `SketchEditor` and add it to a `MapView`.
-1. Use `SketchEditor.start(<SketchCreationMode>)` to start sketching. Modes (as enum) include:
-	* `SketchCreationMode.POINT`
-	* `SketchCreationMode.MULTIPOINT`
-	* `SketchCreationMode.POLYLINE`
-	* `SketchCreationMode.POLYGON`
-	* `SketchCreationMode.FREEHAND_LINE`
-	* `SketchCreationMode.FREEHAND_POLYGON`
-1. Use `.undo()` and `.redo()` on your instance of sketch editor to undo and redo sketch events.
-1. Use `.getGeometry()` on your instance of sketch editor to get the geometry and, for instance, use to to define a new `Graphic`.
-1. Finally, use `.stop()` on your instance of sketch editor to stop sketching.
-
-# Relevant API
+# Sketch editor
+
+Use the Sketch Editor to edit or sketch a new point, line, or polygon geometry on to a map.
+
+![Image of sketch editor App](sketch-editor.png)
+
+## Use case
+
+A field worker could annotate features of interest on a map (via the GUI) such as location of dwellings (marked as points), geological features (polylines), or areas of glaciation (polygons). 
+
+## How to use the sample
+
+Choose which geometry type to sketch from one of the available buttons along the bottom of the screen. Choose from points, multipoints, polylines, polygons, freehand polylines, and freehand polygons.
+
+Use the buttons at the top of the screen to: undo or redo change made to the sketch whilst in sketch mode, and to save the sketch to the graphics overlay.
+
+## How it works
+
+1.  Create a `SketchEditor` and pass it to the MapView with `mapView.setSketchEditor(sketchEditor)`.
+2.  Use `SketchEditor.start(SketchCreationMode)` to start sketching. If editing an existing graphic's geometry, use `SketchEditor.start(graphic.getGeometry)`.
```

Comment:
I think it's worthwhile to keep in, but the sample doesn't seem to have any instances of editing an existing graphic's geometry.

Reply:
good shout: in the Java SE sample we can edit the geometry. I've changed the format here slightly to make it look like an optional comment.